ts/packages/anchor/src/idl.ts

```diff
@@ -67,7 +67,16 @@ export type IdlPda = {
   programId?: IdlSeed;
 };
 
-export type IdlSeed = any; // TODO
+// Based on the rust type defined in lang/sync/src/idl/types.rs and the definitions in lang/syn/src/idl/pda.rs
+export type IdlSeed =
+  | { kind: 'Const'; value: IdlConstant }
+  | { kind: 'Arg'; value: IdlSeedArg }
+  | { kind: 'Account'; value: IdlAccount };
+
+export interface IdlSeedArg {
+  type: IdlType;
+  path: string;
+}
```

Comment:
Hey, thanks for your contribution but I don't think these types are correct based on the Rust definitions you've linked as a comment.

Reply:
Thanks for the feedback. Updated the types, defined separate ones for `IdlSeedConst`, `IdlSeedArg` and `IdlSeedAccount` based on the rust definitions. Let me know the changes required.